.github/workflows/dispatch-build-bottle.yml

```diff
@@ -41,26 +41,38 @@ jobs:
         with:
           script: |
             const intelMacOSRegex = /^\d+(?:\.\d+)?$/;
+            const linuxRegex = /^(ubuntu-|linux-self-hosted-)/;
             return context.payload.inputs.runner.split(",")
                                                 .map(s => s.trim())
                                                 .filter(Boolean)
                                                 .map(s => {
               if (intelMacOSRegex.test(s)) // Ephemeral runners
-                return `${s}-${context.runId}-${process.env.GITHUB_RUN_ATTEMPT}`
+                return {runner: `${s}-${context.runId}-${process.env.GITHUB_RUN_ATTEMPT}`};
+              else if (linuxRegex.test(s))
+                return {
+                  runner:    s,
+                  container: {
+                    image:   "ghcr.io/homebrew/ubuntu22.04:master",
+                    options: "--user=linuxbrew -e GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"
+                  },
+                  workdir:   "/github/home"
+                };
               else
-                return s
+                return {runner: s};
             });
```

Comment:
This is fine to fix things but I'd really love to try and get this amount of logic out of the workflow file and into Homebrew/brew or Homebrew/homebrew-test-bot